src/lib/operators/abstract_aggregate_operator.cpp

```diff
@@ -0,0 +1,51 @@
+#include "abstract_aggregate_operator.hpp"
+#include "abstract_operator.hpp"
+#include "abstract_read_only_operator.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+AbstractAggregateOperator::AbstractAggregateOperator(const std::shared_ptr<AbstractOperator>& in,
+                                                     const std::vector<AggregateColumnDefinition>& aggregates,
+                                                     const std::vector<ColumnID>& groupby_column_ids)
+    : AbstractReadOnlyOperator(OperatorType::Aggregate, in),
+      _aggregates{aggregates},
+      _groupby_column_ids{groupby_column_ids} {
+  Assert(!(aggregates.empty() && groupby_column_ids.empty()),
+         "Neither aggregate nor groupby columns have been specified");
+}
+
+const std::vector<AggregateColumnDefinition>& AbstractAggregateOperator::aggregates() const { return _aggregates; }
+const std::vector<ColumnID>& AbstractAggregateOperator::groupby_column_ids() const { return _groupby_column_ids; }
+
+
+const std::string AbstractAggregateOperator::description(DescriptionMode description_mode) const {
+  std::stringstream desc;
+  desc << "[" << name() << "] " << "GroupBy ColumnIDs: ";
+  for (size_t groupby_column_idx = 0; groupby_column_idx < _groupby_column_ids.size(); ++groupby_column_idx) {
```

Comment:
We are not sure if we want to use the boost library too much, because it results in a strong coupling if we want to replace the library some day. For the metaprogramming it is quite hard to find a alternative, but I think the pure C++ version is readable enough?